### PR TITLE
Popper dot improvements

### PR DESCRIPTION
**Pull request by akashrajr1**
Removes duplicate edges in #637

### CODE REVIEW

**Comment by akashrajr1** [comment]
The commit de57376 addresses #633

**Comment by akashrajr1** [comment]
The commit 8fe74bf addresses #632 
Adds blue border color to workflow node

**Comment by akashrajr1** [comment]
The commit 0f29655 address #634 
Retains original spaces in workflow and action names

File: cli/popper/commands/cmd_dot.py [hunk 1]
```diff
@@ -49,6 +47,16 @@ def add_to_graph(graph_str, wf, parent, children):
...
-        workflow_name = wf.name.replace(' ', '_').replace('-', '_')
-        graph_str = add_to_graph("", wf, workflow_name, wf.root)
-        log.info("digraph G {\n" + graph_str + "}\n")
+        workflow_name = wf.name
+        graph_set = add_to_graph(set(), wf, workflow_name, wf.root)
+        graph_str = '\n'.join(graph_set)
+        workflow_attr = '[ shape=diamond, bordercolor=blue, border=bold]'
+        digraph = '\n'.join(
+            [
+                'digraph G {',
```
Comment (ivotron) [inline cli/popper/commands/cmd_dot.py]: and we can also make the diagram's background transparent by doing:

```
digraph G {
  graph [bgcolor="transparent"]
}
```

File: cli/popper/commands/cmd_dot.py [hunk 2]
```diff
@@ -49,6 +47,16 @@ def add_to_graph(graph_str, wf, parent, children):
     for wfile in wfile_list:
         pipeline = WorkflowRunner(wfile, False, False, False, False, True)
         wf = pipeline.wf
-        workflow_name = wf.name.replace(' ', '_').replace('-', '_')
-        graph_str = add_to_graph("", wf, workflow_name, wf.root)
-        log.info("digraph G {\n" + graph_str + "}\n")
+        workflow_name = wf.name
+        graph_set = add_to_graph(set(), wf, workflow_name, wf.root)
+        graph_str = '\n'.join(graph_set)
+        workflow_attr = '[ shape=diamond, bordercolor=blue, border=bold]'
```
Comment (ivotron) [inline cli/popper/commands/cmd_dot.py]: Can we make this a rounded (transparent) box? Something like:

```
[shape="box", style="filled,rounded", color="crimson", fillcolor="transparent"]
```

this with the goal of having the same symbols (and meaning) as in traditional flowcharts (rounded box denoting the start of a process, and boxes denoting subprocesses). The use of `cyan` would make the diagram amenable to both dark and light terminal backgrounds (assuming something like [`imgcat`](https://github.com/eddieantonio/imgcat#build) is being used)

File: cli/popper/commands/cmd_dot.py [hunk 3]
```diff
@@ -27,17 +27,15 @@ def cli(ctx, wfile, recursive):
...
     """
-    def add_to_graph(graph_str, wf, parent, children):
+    def add_to_graph(graph_set, wf, parent, children):
         """Recursively goes through "next" and adds corresponding actions
         """
-        _parent = parent.replace(' ', '_').replace('-', '_')
         for n in children:
-            _n = n.replace(' ', '_').replace('-', '_')
-            graph_str += "  {} -> {};\n".format(_parent, _n)
+            graph_set.add('  "{}" -> "{}";'.format(parent, n))
```
Comment (ivotron) [inline cli/popper/commands/cmd_dot.py]: can we also make this a transparent box? something like:

```
[shape="box", style="filled", color="cyan", fillcolor="transparent"]
```

`cyan` seems to be a color that can be visualized on both dark and light terminal background colors.

**Comment by ivotron** [comment]
closing in favor of #655

**Pull request closed without merging**